Update docs and CI to support rabbitmq-external chart (#655)
wireapp/wire-server-deploy: offline/federation_preparation.md

## RabbitMQ

> There are two ways to deploy the RabbitMQ cluster:

Reviewer: ```suggestion
There are two methods to deploy the RabbitMQ cluster:
```

One general rule of technical writing is to use one name for one thing. That's usually easier to grasp for readers. So, you should describe here if we're talking about ways or methods.

> Method 1

Reviewer: The heading declarations are strange here: See inline fixes.

---
## Install RabbitMQ inside kubernetes cluster with the help of helm chart

> To install the rabbitmq,
> First copy the value and secret file:

Reviewer: ```suggestion
To install the RabbitMQ service, first copy the value and secret file:
```

```
cp ./values/rabbitmq/prod-values.example.yaml ./values/rabbitmq/values.yaml
cp ./values/rabbitmq/prod-secrets.example.yaml ./values/rabbitmq/secrets.yaml
```

> By default this will use ephemeral storage, to use the local persistance storage in kubernetes nodes, please refer to this doc - offline/local_persistance_storage.md
> 

Reviewer: ```suggestion
By default this will create a RabbitMQ deployment with ephemeral storage. To use the local persistence storage of Kubernetes nodes, please refer to the related documentation in [offline/local_persistance_storage.md](./offline/local_persistance_storage.md).

```

If the filename contains the typo *persistance* (vs. *persistence*), too, it should be renamed.

> By default this will use ephemeral storage, to use the local persistance storage in kubernetes nodes, please refer to this doc - offline/local_persistance_storage.md
> 

Reviewer: ```suggestion
By default this will create a RabbitMQ deployment with ephemeral storage. To use the local persistence storage of Kubernetes nodes, please refer to the related documentation in [offline/local_persistance_storage.md](./offline/local_persistance_storage.md).

```

If the filename contains the typo *persistance* (vs. *persistence*), too, it should be renamed.

> Now, update the `./values/rabbitmq/values.yaml` and `./values/rabbitmq/secrets.yaml` with correct values as per needed.

Reviewer: ```suggestion
Now, update the `./values/rabbitmq/values.yaml` and `./values/rabbitmq/secrets.yaml` with correct values as needed.
```

> Deploy the rabbitmq helm chart -

Reviewer: ```suggestion
Deploy the rabbitmq helm chart:
```

```
d helm upgrade --install rabbitmq ./charts/rabbitmq --values ./values/rabbitmq/values.yaml --values ./values/rabbitmq/secrets.yaml
```

> Method 2
> ---
> 
> ## Installing RabbitMQ outside kubernets cluster with ansible playbook

Reviewer: ```suggestion
### Method 2: Installing RabbitMQ outside kubernets cluster with ansible playbook
```

> Add the nodes in which you want to run rabbitmq to the `[rmq-cluster]` group in the `ansible/inventory/offline/hosts.ini` file. Also, update the `ansible/roles/rabbimq-cluster/defaults/main.yml` file with the correct configurations for your environment.

Reviewer: ```suggestion
Add the nodes in which you want to run RabbitMQ to the `[rmq-cluster]` group in the `ansible/inventory/offline/hosts.ini` file. Also, update the `ansible/roles/rabbimq-cluster/defaults/main.yml` file with the correct configurations for your environment.
```

> Add the nodes in which you want to run rabbitmq to the `[rmq-cluster]` group in the `ansible/inventory/offline/hosts.ini` file. Also, update the `ansible/roles/rabbimq-cluster/defaults/main.yml` file with the correct configurations for your environment.

Reviewer: ```suggestion
Add the nodes on which you want to run rabbitmq to the `[rmq-cluster]` group in the `ansible/inventory/offline/hosts.ini` file. Also, update the `ansible/roles/rabbimq-cluster/defaults/main.yml` file with the correct configurations for your environment.
```

> You should have following entries in the /ansible/inventory/offline/hosts.ini file for example:

Reviewer: ```suggestion
You should have following entries in the `/ansible/inventory/offline/hosts.ini` file. For example:
```

> You should have following entries in the /ansible/inventory/offline/hosts.ini file for example:

Reviewer: An explanation what these *entries* mean would be great. `rabbitmq_network_interface` should be pretty clear. But, how do I figure out `ansnode1` et. al. for my setup?

```
[rmq-cluster:vars]
rabbitmq_network_interface = enp1s0

[rmq-cluster]
ansnode1
ansnode2
ansnode3
```

> Important: RabbitMQ nodes address each other using a node name, for e.g rabbitmq@ansnode1

Reviewer: ```suggestion
*Important:* RabbitMQ nodes address each other using a node name, for e.g rabbitmq@ansnode1
```

Reviewer: If it's important, set it in bold face to make sure it's not easily missing :wink:

> Important: RabbitMQ nodes address each other using a node name, for e.g rabbitmq@ansnode1

Reviewer: ```suggestion
Important: RabbitMQ nodes address each other using a node name, e.g. `rabbitmq@ansnode1`.
```

> Please refer to official doc and configure your DNS based on the setup - https://www.rabbitmq.com/clustering.html#cluster-formation-requirements

Reviewer: ```suggestion
Please refer to the official documentation and configure your DNS based on the setup - https://www.rabbitmq.com/clustering.html#cluster-formation-requirements
```

> For adding entries to local host file(/etc/hosts), run

Reviewer: ```suggestion
For adding entries to local host file(`/etc/hosts`), run
```

```
d ansible-playbook -i ansible/inventory/offline/hosts.ini ansible/roles/rabbitmq-cluster/tasks/configure_dns.yml
```

> Create the rabbitmq cluster - 

Reviewer: ```suggestion
Create the rabbitmq cluster:
```

> Create the rabbitmq cluster - 

Reviewer: ```suggestion
Create the rabbitmq cluster:
```

``` 
d ansible-playbook -i ansible/inventory/offline/hosts.ini ansible/rabbitmq.yml
```

> Make kubernetes aware of where RabbitMQ external stateful service is running:

Reviewer: ```suggestion
Make Kubernetes aware of where RabbitMQ external stateful service is running:
```

```
d helm install rabbitmq-external ./charts/rabbitmq-external --values ./values/rabbitmq-external/values.yaml
```

> Configure wire-server to use the external rabbitmq service:

Reviewer: ```suggestion
Configure wire-server to use the external RabbitMQ service:
```

To keep the text consistent, it would be helpful to decide when to use which way of writing "RabbitMQ". IMHO *RabbitMQ* should be used when you're writing about the technology in general and *rabbitmq* when you're referring to the Helm thing.

> Edit the `/values/wire-server/prod-values.yaml` file to update the rabbitmq host

Reviewer: ```suggestion
Edit the `/values/wire-server/prod-values.yaml` file to update the RabbitMQ host
```

> Under `brig` and `galley` section, you will find the "rabbitmq" config, update the host to "rabbitmq-external", it should look like this:

Reviewer: ```suggestion
Under `brig` and `galley` section, you will find the `rabbitmq` config, update the host to `rabbitmq-external`, it should look like this:
```

Let's stay consistent with the usage of monospaced fonts. I.e. let's put everything that shows up literally in code in backticks.

Now, update the `./values/rabbitmq/values.yaml` and `./values/rabbitmq/secrets.yaml` with correct values as needed.

Deploy the `rabbitmq` helm chart:
```
d helm upgrade --install rabbitmq ./charts/rabbitmq --values ./values/rabbitmq/values.yaml --values ./values/rabbitmq/secrets.yaml
```

### Method 2: Install RabbitMQ outside of the Kubernetes cluster with an Ansible playbook

Add the nodes on which you want to run rabbitmq to the `[rmq-cluster]` group in the `ansible/inventory/offline/hosts.ini` file. Also, update the `ansible/roles/rabbitmq-cluster/defaults/main.yml` file with the correct configurations for your environment.

If you need RabbitMQ to listen on a different interface than the default gateway, set `rabbitmq_network_interface`

You should have following entries in the `/ansible/inventory/offline/hosts.ini` file. For example:
```
[rmq-cluster:vars]
rabbitmq_network_interface = enp1s0

[rmq-cluster]
ansnode1
ansnode2
ansnode3
```

> **Important:** RabbitMQ nodes address each other using a node name, for e.g rabbitmq@ansnode1

Author: the link below this line in the doc describes all the requirements and naming conventions

For adding entries to local host file(`/etc/hosts`), run
```
d ansible-playbook -i ansible/inventory/offline/hosts.ini ansible/roles/rabbitmq-cluster/tasks/configure_dns.yml
```

Create the rabbitmq cluster:

``` 
d ansible-playbook -i ansible/inventory/offline/hosts.ini ansible/rabbitmq.yml
```

and run the following playbook to create values file for helm charts to look for RabbitMQ IP addresses -

```
d ansible-playbook -i ./ansible/inventory/offline/hosts.ini ansible/helm_external.yml --tags=rabbitmq-external
```

Make Kubernetes aware of where RabbitMQ external stateful service is running:
```
d helm install rabbitmq-external ./charts/rabbitmq-external --values ./values/rabbitmq-external/values.yaml
```

Configure wire-server to use the external RabbitMQ service:

Edit the `/values/wire-server/prod-values.example.yaml` file to update the RabbitMQ host
Under `brig` and `galley` section, you will find the `rabbitmq` config, update the host to `rabbitmq-external`, it should look like this:
```
rabbitmq:
  host: rabbitmq-external

> ``` 

Author: re-installing is not required. 
According to the docs, this step is before deploying wire-server
